jmclient/jmclient/wallet_utils.py

```diff
@@ -399,16 +399,22 @@ def get_addr_status(addr_path, utxos, is_new, is_internal):
             if addr_path != utxodata['path']:
                 continue
             addr_balance += utxodata['value']
-            is_coinjoin, cj_amount, cj_n = \
-                get_tx_info(binascii.hexlify(utxo[0]).decode('ascii'))[:3]
-            if is_coinjoin and utxodata['value'] == cj_amount:
-                status.append('cj-out')
-            elif is_coinjoin:
-                status.append('change-out')
-            elif is_internal:
-                status.append('non-cj-change')
-            else:
-                status.append('deposit')
+            #TODO it is a failure of abstraction here that
+            # the bitcoin core interface is used directly
```

Comment:
Right! And while adding yet more features is always a cognitive load, one very nice thing about you going through this exercise of 'add new bci' is you find this.

Reply:
And I think I know why this situation happened, the code was originally only used in wallet-tool.py history, which checked that Core was being used and raised an error if not. Then this part of the code was clearly copypasted here and used always even if the blockchain interface wasnt Bitcoin Core.